cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/impl/CassandraSchemaDefiner.java

```diff
@@ -101,18 +102,36 @@ private void processIndexes(CassandraDatastoreProvider datastoreProvider, Table
 		while ( fkMappings.hasNext() ) {
 			ForeignKey foreignKey = fkMappings.next();
 
-			List<String> fkColumnNames = new ArrayList<String>();
-
-			Iterator<Column> fkColumnIterator = foreignKey.getColumnIterator();
-			while ( fkColumnIterator.hasNext() ) {
-				Column column = fkColumnIterator.next();
-				fkColumnNames.add( column.getName() );
-			}
+			List<String> fkColumnNames = extractColumnNames( foreignKey.getColumnIterator() );
 
 			// cassandra won't allow single index on multiple cols, so index first col only.
 			if ( !primaryKeys.contains( fkColumnNames.get( 0 ) ) ) {
 				datastoreProvider.createSecondaryIndexIfNeeded( table.getName(), fkColumnNames.get( 0 ) );
 			}
 		}
+
+
+		Iterator<Index> indexIterator = table.getIndexIterator();
+		while ( indexIterator.hasNext() ) {
+			Index index = indexIterator.next();
+
+			List<String> columnNames = extractColumnNames( index.getColumnIterator() );
+
+			// cassandra won't allow single index on multiple cols, so index first col only.
+			if ( !primaryKeys.contains( columnNames.get( 0 ) ) ) {
```

Comment:
We should raise a warning if a multi-column index is configured which cannot be mapped.